mission/mission_core/missions/ScanTheCode_mission.py

```diff
@@ -0,0 +1,249 @@
+        '''
+        Camera data is a dictionary with the following format:
+        {
+            "port": CameraData,
+            "starboard": CameraData,
+            "center": CameraData
+        }
+        CameraData is a class that contains the image frame and the results of a Yolo model (if running).
+        And can be accessed like so:
+            camera_data.frame
+            camera_data.results
+
+        PositionData is a class that contains the current position and heading of the vehicle.
+        And can be accessed like so:
+            position_data.lat : float
+            position_data.lon : float
+            position_data.position : tuple (lat, lon)
+            position_data.heading : float
+
+        The output perc_cmd has the following format (does not have to be filled):
+        {
+            "start": ["camera_name"],
+            "stop": ["camera_name"],
+            "record": ["camera_name"],
+            "stop_record": ["camera_name"],
+            "load_model": [("camera_name", "model_path")],
+            "stop_model": ["camera_name"]
+        }
+
+        The output gnc_cmd has the following format (does not have to be filled):
+        {
+            "heading": float, # The target heading for the boat to turn towards
+            "vector": tuple, # The target vector for the boat to head towards (will override heading)
+            "poshold": bool, # Whether to set the GNC to poshold mode (heading is not maintained but position is)
+            "waypoint": tuple, # The target waypoint to set the GNC to (sending any other commands will override this)
+            "end_mission": bool # Whether to end the mission
+        }
+
+        '''
+        
+        # Run one yolo inference on the provided frame at camera_data.frame
+        self.storageArray = camera_data.results[0]
+        
+        if(len(self.storageArray) != 0):
+            if(self.debug):
+                print("Interim processing")
+                print(self.storageArray)
+            returnVal = filter(self.storageArray)
+            records = self.analyze(returnVal)
+
+            self.colors, self.pattern = self.results(records)
+            if(self.debug):
+                print("printing results")
+                print(self.colors[0] + self.colors[1] + self.colors[2])
+                print("printing dict outside")
+                print(records)
+            
+            max_vote = records.get(self.pattern)
+            if max_vote is None:
+                max_vote = 0
+            # if(max_vote > 0):
+            #     # Want to update colors for heartbeat
+            #     update_colors_on_server(colors, max_vote, len(storageArray))
+            # else:
+            #     # No Colors to update, can just say black
+            #     update_colors_on_server(["Black", "Black", "Black"], 0, len(storageArray))
+        
+        
+        
+        
+        perc_cmd = {
```

Comment:
Only use perc_cmd to make changes to the current perception config. It doesn't need to be sent every time. (remember this run method is run in a loop). So in your case I would just leave it as an empty dict.

Reply:
Good to know it doesn't need to be sent everytime.
Resolved